src/main/scala/com/gu/fulfilmentLookup/BasicAuth.scala

```diff
@@ -0,0 +1,29 @@
+package com.gu.fulfilmentLookup
+
+object BasicAuth extends Logging {
+
+  def validAuth(auth: Option[String], config: Config): Boolean = {
```

Comment:
actually most of this code is doing "boring" work so we can try to minimise it.
can create a `case class AuthDetails(user: String, pass: String)` instead of the tuple.
then we can make decodeBasicAuth return one (or an Option of one?)
we can also pass one in instead of `config`
then this whole function becomes
`def isValidAuth(auth: Option[String], configAuthDetails: AuthDetails): Boolean = auth.map(decodeBasicAuth).exists(configAuthDetails)`